cmd/argocd-util/main.go

```diff
@@ -154,6 +162,77 @@ func NewGenDexConfigCommand() *cobra.Command {
 	return &command
 }
 
+// NewExportCommand defines a new command for exporting Kubernetes and Argo CD resources.
+func NewExportCommand() *cobra.Command {
+	var (
+		clientConfig clientcmd.ClientConfig
+		out          string
+		clientOpts   argocdclient.ClientOptions
+	)
+	var command = cobra.Command{
+		Use:   "export",
+		Short: "Export all Argo CD data",
+		RunE: func(c *cobra.Command, args []string) error {
+			config, err := clientConfig.ClientConfig()
+			errors.CheckError(err)
+			namespace, _, err := clientConfig.Namespace()
+			errors.CheckError(err)
+			kubeClientset := kubernetes.NewForConfigOrDie(config)
+
+			settingsMgr := settings.NewSettingsManager(kubeClientset, namespace)
+			settings, err := settingsMgr.GetSettings()
+			errors.CheckError(err)
+			settingsData, err := yaml.Marshal(settings)
+			errors.CheckError(err)
+
+			db := db.NewDB(namespace, kubeClientset)
+			clusters, err := db.ListClusters(context.Background())
+			errors.CheckError(err)
+			clusterData, err := yaml.Marshal(clusters)
+			errors.CheckError(err)
+
+			repos, err := db.ListRepositories(context.Background())
+			errors.CheckError(err)
+			repoData, err := yaml.Marshal(repos)
+			errors.CheckError(err)
+
+			conn, appIf := argocdclient.NewClientOrDie(&clientOpts).NewApplicationClientOrDie()
+			defer util.Close(conn)
+			apps, err := appIf.List(context.Background(), &application.ApplicationQuery{})
+			errors.CheckError(err)
+			appsData, err := yaml.Marshal(apps)
+			errors.CheckError(err)
+
+			outputStrings := []string{
+				string(settingsData),
+				string(repoData),
+				string(clusterData),
+				string(appsData),
+			}
+			output := fmt.Sprint(strings.Join(outputStrings, "\n---\n"))
+
+			if out == "" {
+				fmt.Println(output)
+			} else {
+				err = ioutil.WriteFile(out, []byte(output), 0644)
+				errors.CheckError(err)
+			}
+			return nil
+		},
+	}
+
+	clientConfig = cli.AddKubectlFlagsToCmd(&command)
+	command.Flags().StringVarP(&out, "out", "o", "", "Output to the specified file instead of stdout")
+	command.Flags().StringVar(&clientOpts.ConfigPath, "config", "", "Path to ArgoCD config")
```

Comment:
Looks like clientOpts are not required , since command is talking to kubernetes directly. Please remove

Reply:
@alexmt I was under the impression I needed this to export apps, e.g.,:

```
			conn, appIf := argocdclient.NewClientOrDie(&clientOpts).NewApplicationClientOrDie()
			defer util.Close(conn)
			apps, err := appIf.List(context.Background(), &application.ApplicationQuery{})
			errors.CheckError(err)
			appsData, err := yaml.Marshal(apps)
			errors.CheckError(err)
```

Reply:
Instead of using server and client please use application client directly:

```
s.appclientset.ArgoprojV1alpha1().Applications(s.ns).List(metav1.ListOptions{})
```